cogito/putter.go

```diff
@@ -81,6 +91,13 @@ func (putter *ProdPutter) ProcessInputDir() error {
 
 	params := putter.Request.Params
 	source := putter.Request.Source
+
+	// Validate optional sinks configuration.
+	sinks, err := MergeAndValidateSinks(source.Sinks, params.Sinks)
```

Comment:
Why the hell do we have to recompute, revalidate and rehandle the error once more in `ProcessInputDir` ?! All this already took place in `LoadConfiguration` and it is always called before `ProcessInputDir` and nothing could have changed in between. Is that just to avoid saving the merged sinks in `putter`? Why not?

Reply:
Because it doesn't work that way. As I said during the stand-up, I spent almost the whole day trying to figure out why `putter.Sinkers = putter.Sinks()` ended having the sinkers always empty breaking all the code. I really tried to have this object set just once and then available, but it didn't work.

Reply:
So I finally gave it a try to understand what you were talking about. Basically, in itself, it just works as I expected.
The actual issue is with the tests. Both `ProcessInputDirs` and `Sinks` now require a bit more state to be setup as `putter.sinks` is set by `LoadConfiguration`.
I just pushed a branch on top of yours that does just that: `refactor_putter_tests`. Might not be the best implementation however. So to be discussed. @marco-m-pix4d, can you have a look please?

Reply:
Meanwhile I removed the check in `ProcessInputDir()`, we want just the sinks there.

Reply:
@odormond, before I take a look at the branch `refactor_putter_tests` (which might solve everything for the better for what I know), a more high level summary of the pair programming that Andrea and myself did Wednesday is the following:

- Yes, there is code duplication.
- On the other hand, the concern I had with your proposal is that it puts more state in the Putter object for the sake of reducing code duplication. This is a trade-off that is not always worth it: as you mention when referring to `refactor_putter_tests`, it requires to add more state in the tests.
- On the other hand, I am deep into Consul and I find it difficult, in these days, to put myself with enough depth in this code to follow all changes, so maybe I am making a mountain out of a mole hill.
- I think it is quite important to merge this PR so that we start having the feature and see how it behaves "for real". As long as we do not cut a release, nobody will pick it up in a pipeline in any case.

Would it be OK for you to proceed with merging this PR, and consider branch `refactor_putter_tests` as a follow-up PR? Or am I missing something that fundamental that we want to do only one merge?